multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -193,11 +216,37 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws ServerException {
 
   @VisibleForTesting
   void checkRamResourcesAvailability(
-      String accountId, String namespace, WorkspaceConfig config, @Nullable String envName)
+      String accountId, String namespace, Workspace workspace, @Nullable String envName)
+      throws NotFoundException, ServerException, ConflictException {
+    if (workspace.getConfig() != null) {
+      checkRamResourcesAvailability(
+          accountId, namespace, workspace.getId(), workspace.getConfig(), envName);
+    } else {
+      LOG.warn(
+          "Can not estimate memory needed for devfile based workspace {}. Set memory resources limits may work incorrectly",
```

Comment:
How about: ``"Cannot estimate the memory requirements of a devfile-based workspace `{}`. The memory limits may not be applied correctly."``